tests/catalog/test_sql.py

```diff
@@ -225,6 +237,93 @@ def test_creation_from_impl(catalog_name: str, warehouse: Path) -> None:
     )
 
 
+def confirm_no_tables_exist(alchemy_engine: Engine) -> list[str]:
+    inspector = inspect(alchemy_engine)
+    for c in SqlCatalogBaseTable.__subclasses__():
+        if inspector.has_table(c.__tablename__):
+            c.__table__.drop(alchemy_engine)
+
+    catalog_tables = [c.__tablename__ for c in SqlCatalogBaseTable.__subclasses__()]
+    any_table_exists = any(t for t in inspector.get_table_names() if t in catalog_tables)
+    if any_table_exists:
+        pytest.raises(TableAlreadyExistsError, "Tables exist, but should not have been created yet")
+
+    return catalog_tables
+
+
+def confirm_all_tables_exist(inspector: Inspector, catalog_tables: list[str], catalog: Catalog) -> None:
```

Comment:
It looks like we are passing the `catalog` here just to assert its type - is this necessary for this check? Was the intention to check the tables through the catalog's engine instead like:

```
inspect(catalog.engine).get_table_names()
```

Reply:
Not passing the inspector anymore. Also needed to cast the Catalog to use the engine. Made the uri a fixture to insure using same source between catalog and inspector that was creating tables.